samcli/lib/build/app_builder.py

```diff
@@ -151,14 +160,46 @@ def update_template(self, template_dict, original_template_path, built_artifacts
 
             resource_type = resource.get("Type")
             properties = resource.setdefault("Properties", {})
-            if resource_type == "AWS::Serverless::Function":
+            if resource_type == SamBaseProvider.SERVERLESS_FUNCTION:
                 properties["CodeUri"] = artifact_relative_path
 
-            if resource_type == "AWS::Lambda::Function":
+            if resource_type == SamBaseProvider.LAMBDA_FUNCTION:
                 properties["Code"] = artifact_relative_path
 
+            if resource_type in [SamBaseProvider.SERVERLESS_LAYER, SamBaseProvider.LAMBDA_LAYER]:
+                properties["ContentUri"] = artifact_relative_path
+
         return template_dict
 
+    def _build_layer(self, layer_name, codeuri, runtime):
+        # Create the arguments to pass to the builder
+        # Code is always relative to the given base directory.
+        code_dir = str(pathlib.Path(self._base_dir, codeuri).resolve())
+
+        config = get_workflow_config(runtime, code_dir, self._base_dir)
+        subfolder = get_layer_subfolder(runtime)
+
+        # artifacts directory will be created by the builder
+        artifacts_dir = str(pathlib.Path(self._build_dir, layer_name, subfolder))
+
+        with osutils.mkdir_temp() as scratch_dir:
```

Comment:
something to check that this works as well on windows as well. I know we have a utility for platform independent temp file, but thats for temp files.

Reply:
I'll validate on windows, didn't do that.

Reply:
Validated on windows: https://pastebin.com/Sj6ZwbZr